Packs/SymantecEndpointSecurity/Integrations/SymantecEndpointSecurity/SymantecEndpointSecurity.py

```diff
@@ -0,0 +1,459 @@
+
+    Returns:
+        str: The normalized date string without milliseconds.
+    """
+    try:
+        # Parse the original date string with milliseconds
+        original_date = datetime.strptime(date_str, "%Y-%m-%dT%H:%M:%S.%fZ")
+    except Exception:
+        if "." in date_str:
+            date_str = f"{date_str.split('.')[0]}Z"
+        return date_str
+
+    # Convert back to the desired format without milliseconds
+    new_date_str = original_date.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    return new_date_str
+
+
+def update_new_integration_context(
+    filtered_events: list[dict[str, str]],
+    next_hash: str,
+    include_last_fetch_events: bool,
+    last_integration_context: dict[str, str],
+):
+    """
+    Update the integration context.
+
+    Args:
+        filtered_events (list[dict[str, str]]): A list of filtered events.
+        next_hash (str): The hash for the next fetch operation.
+        include_last_fetch_events (bool): Flag to include last fetched events in the integration context.
+        last_integration_context (dict[str, str]): The previous integration context.
+    """
+    events_suspected_duplicates = (
+        extract_events_suspected_duplicates(filtered_events) if filtered_events else []
+    )
+
+    # Determine the latest event time: Extract the last time of the filtered event,
+    # or use the last known time from integration context if no filtered events exist
+    latest_event_time = (
+        normalize_date_format(max(filtered_events, key=parse_event_time)["log_time"])
+        if filtered_events
+        else last_integration_context.get("latest_event_time", "")
+    )
+
+    # If the latest event time matches the previous one,
+    # extend the suspected duplicates list with events from the previous context,
+    # to control deduplication across multiple fetches.
+    if latest_event_time == last_integration_context.get("latest_event_time", ""):
+        events_suspected_duplicates.extend(
+            last_integration_context.get("events_suspected_duplicates", [])
+        )
+
+    integration_context = {
+        "latest_event_time": latest_event_time,
+        "events_suspected_duplicates": events_suspected_duplicates,
+        "next_fetch": {"next": next_hash} if next_hash else {},
+        "last_fetch_events": filtered_events if include_last_fetch_events else [],
+    }
+
+    demisto.info(f"Updating integration context with new data: {integration_context=}")
+    set_integration_context(integration_context)
+
+
+def push_events(events: list[dict]):
+    """
+    Push events to XSIAM.
+    """
+    send_events_to_xsiam(events=events, vendor=VENDOR, product=PRODUCT)
+    demisto.info(f"{len(events)} events were pushed to XSIAM")
+
+
+def parse_event_time(event) -> datetime:
+    """
+    Parse the event time from the given event dict to datetime object.
+    """
+    return datetime.strptime(
+        normalize_date_format(event["log_time"]), "%Y-%m-%dT%H:%M:%SZ"
+    )
```

Comment:
since it's only used in `extract_events_suspected`, move it into it (or just make it a lambda as the `key` there)